app/src/main/java/ch/nevis/exampleapp/coroutines/dagger/ApplicationModule.kt

```diff
@@ -291,41 +303,45 @@ class ApplicationModule {
     fun provideInBandAuthenticationUseCase(
         clientProvider: ClientProvider,
         stateRepository: OperationStateRepository<UserInteractionOperationState>,
-        errorHandlerChain: ErrorHandlerChain,
         pinUserVerifier: PinUserVerifier,
         fingerprintUserVerifier: FingerprintUserVerifier,
         biometricUserVerifier: BiometricUserVerifier,
-        settings: Settings
+        devicePasscodeUserVerifier: DevicePasscodeUserVerifier,
+        settings: Settings,
+        onError: Consumer<AuthenticationError>
     ): InBandAuthenticationUseCase = InBandAuthenticationUseCaseImpl(
         clientProvider,
         stateRepository,
         provideAuthenticationAuthenticatorSelector(stateRepository, settings),
         pinUserVerifier,
         fingerprintUserVerifier,
         biometricUserVerifier,
+        devicePasscodeUserVerifier,
         provideOnSuccessAuthentication(stateRepository),
-        provideOnErrorForUserInteractionOperation(stateRepository, errorHandlerChain)
```

Comment:
Fixed. You are right it was inconsistent. The reason why provide method call is used is that the return value is the same e.g. in case of `provideOnSuccessForUserInteractionOperation` and `provideOnSuccessForChangePinOperation`. Without the method call the injected value could be ambiguous. However in case of error callbacks it is not necessary, so I removed all method calls for them.